pkg/actions/lua/lakefs/catalogexport/delta_exporter.lua

```diff
@@ -0,0 +1,134 @@
+                    {"remove":{"path":"part-00000-d660b401-ceec-415a-a791-e8d1c7599e3d-c000.snappy.parquet","deletionTimestamp":1699276565259,"dataChange":true,"extendedFileMetadata":true,"partitionValues":{},"size":82491}}
+            ]]
+            for _, e in ipairs(content) do
+                local entry = json.unmarshal(e)
+                local p = ""
+                if entry.add ~= nil then
+                    p = entry.add.path
+                elseif entry.remove ~= nil then
+                    p = entry.remove.path
+                end
+                if p ~= "" then
+                    local code, obj = lakefs.stat_object(repo, commit_id, pathlib.join("/",path, p))
+                    if code == 200 then
+                        local obj_stat = json.unmarshal(obj)
+                        local physical_path = obj_stat["physical_address"]
+                        if entry.add ~= nil then
+                            entry.add.path = physical_path
+                        elseif entry.remove ~= nil then
+                            entry.remove.path = physical_path
+                        end
+                    end
+                end
+                local entry_m = json.marshal(entry)
+                table.insert(entry_log, entry_m)
+            end
+            table_log[string.format("%s.json", keyGenerator())] = entry_log
+        end
+
+        -- Get the table delta log physical location
+        local t_name = pathlib.parse(path)["base_name"]
+        local table_export_prefix = utils.get_storage_uri_prefix(ns, commit_id, action)
+        local table_physical_path = pathlib.join("/", table_export_prefix, t_name)
+        local table_log_physical_path = pathlib.join("/", table_physical_path, "_delta_log")
+
+        -- Upload the log to this physical_address
+        local storage_props = utils.parse_storage_uri(table_log_physical_path)
+        --[[
+            table_log:
+                {
+                    <version1>.json : [
+                        {"commitInfo":...}, {"add": ...}, {"remove": ...},...
+                    ],
+                    <version2>.json : [
+                        {"commitInfo":...}, {"add": ...}, {"remove": ...},...
+                    ],...
+                }
+        ]]
+        for entry_version, table_entry in pairs(table_log) do
+            local table_entry_string = ""
+            for _, content_entry in ipairs(table_entry) do
+                table_entry_string = table_entry_string .. content_entry
+            end
+            local version_key = storage_props.key .. "/" .. entry_version
+            storage_client.put_object(storage_props.bucket, version_key, table_entry_string)
```

Comment:
Suggestion for api consistency - **feel free to decide it's irrelevant in this case and discard**.

When troubleshooting the flow as a user, you don't always want to write to actual storage every time. 
A client can of course add a mock implementation of `storage_client` in that case, so that's good. 
But, in order to be **consistent with all exporters**, consider adding `options` parameter and use it like the [symlink_exporter.lua](https://github.com/treeverse/lakeFS/blob/e5ca6b86baa2c38d2c3ddaaca0db3b014d15906f/pkg/actions/lua/lakefs/catalogexport/symlink_exporter.lua#L92)

```lua
local function export_delta_log(action, table_paths, storage_client, delta, options)
   local put_object = opts.writer or storage_client.put_object
```

Reply:
Don't see the advantage of injecting a writer in two different ways to the same function, yet I do see the advantage of generalizing the concept of a writer. 
I'll change the `storage_client` object to a writer function instead.